CMakeLists.txt

```diff
@@ -32,12 +32,22 @@ endif()
 
 set(BIN_DIRECTORY bin)
 
-set(OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}/${BIN_DIRECTORY})
+set(OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/${BIN_DIRECTORY})
 
 #set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${OUTPUT_DIRECTORY})
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${OUTPUT_DIRECTORY})
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${OUTPUT_DIRECTORY})
 
+# Remove when https://gitlab.kitware.com/cmake/cmake/issues/18525 is addressed
+set(OPENSSL_ENGINES_DIR "" CACHE PATH "OpenSSL Engines Directory")
+if ("${OPENSSL_ENGINES_DIR}" STREQUAL "")
+	include(FindPkgConfig)
```

Comment:
does cmake at travis supports that ? it must be pretty recent CMake, right ?

Reply:
It doesn't support https://gitlab.kitware.com/cmake/cmake/issues/18525 because cmake hasn't done anything with that issue yet :-) The comment is just a reminder that some day, if that issue is fixed in cmake, to remove this block as it would then be unnecessary.

The code that's here is supported by cmake including that on Travis. Note that `OPENSSL_ENGINE_DIR` is explicitly specified in .travis.yml since we want to use the OpenSSL version we built and not the one included in the Travis provided OS.